shipyard-controller/nats/connection_test.go

```diff
@@ -77,6 +77,54 @@ func TestNatsConnectionHandler(t *testing.T) {
 	}, 15*time.Second, 5*time.Second)
 }
 
+func TestNatsConnectionHandler_ShutdownSubscriber(t *testing.T) {
+	mockNatsEventHandler := &natsmock.IKeptnNatsMessageHandlerMock{
+		ProcessFunc: func(event apimodels.KeptnContextExtendedCE, sync bool) error {
+			return nil
+		},
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+
+	nh := NewNatsConnectionHandler(ctx, natsURL())
+
+	err := nh.SubscribeToTopics([]string{"sh.keptn.>"}, NewKeptnNatsMessageHandler(mockNatsEventHandler.Process))
+
+	require.Nil(t, err)
+
+	publisher, err := nh.GetPublisher()
+
+	require.Nil(t, err)
+
+	event := cloudevents.NewEvent()
+	event.SetType(keptnv2.GetTriggeredEventType("test"))
+	_ = event.SetData(cloudevents.ApplicationJSON, map[string]interface{}{
+		"project": "my-project",
+	})
+
+	// send with Send method
+	err = publisher.Send(context.TODO(), event)
+	require.Nil(t, err)
+
+	// send with SendEvent method
+	err = publisher.SendEvent(event)
+	require.Nil(t, err)
+
+	require.Eventually(t, func() bool {
+		return len(mockNatsEventHandler.ProcessCalls()) == 2
```

Comment:
why didn't we use `require.Equal(t, 2, len(mockNatsEventHandler.ProcessCalls()))` as on line 125 ?

Reply:
exaplained, looks good then